Eastern-Research-Group/mywaterway
# Feature/hmw 47 plan summary virtualized list

## Related Issues:
* https://jira.epa.gov/browse/HMW-47

## Main Changes:
* Converted the list under "Waters Covered" of the plan summary page to a virtualized list for better performance.
  * Created a react-virtualized wrapper component (VirtualizedList) as part of this. Also updated the State Waterbody List component to use this new wrapper component. This wrapper component will make it easier to use the virtualized list in the future.
* Converted styles in the state waterbody list, actions, and waterbody report components to use the css property instead of styled components. I went a head and converted the waterbody report since it was very similar styles to the actions page, also it allowed me to go ahead and remove some of the styled components that were being imported.

## Steps To Test:
1. Navigate to http://localhost:3000/plan-summary/WIDNR/UpperFoxWolf_Protection2020
2. Scroll down the list on the right 
3. Verify the performance is better than the performance on production
4. Verify the expand/collapse all button works
5. Verify expanding/collapsing individual items works
6. Navigate to http://localhost:3000/state/AL/advanced-search
7. Click search
8. Switch to the list view
9. Repeat steps 2, 4 and 5

## TODO:
- [ ] Update styles on the actions page. @courtneymyers You had mentioned the styles need updating. Would you like to do that as part of this PR or a separate one? Feel free to commit to this branch if you want it included with this.
- [x] Switch from using the `StyledErrorBox` component to using `errorBoxStyles`. The `feature/panel-changes-mockup` branch needs to be merged before we can make this switch.
